src/CMakeLists.txt

```diff
@@ -169,8 +171,18 @@ endif()
 
 
 install( DIRECTORY ../examples/
-         DESTINATION examples )
+         DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/${CMAKE_PROJECT_NAME}/examples )
 
 install( DIRECTORY ../scripts/
-         DESTINATION scripts )
-
+         DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/${CMAKE_PROJECT_NAME}/scripts )
+
+add_custom_target( geosx_generate_install_schema
+         ALL
+         COMMAND geosx -s ${CMAKE_BINARY_DIR}/schema.xsd >generate_schema.log 2>&1 || (cat generate_schema.log && exit 1)
+         WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+         DEPENDS geosx
+         COMMENT "Generating XML schema to install"
+         )
```

Comment:
I confess I was kind of fearing this: we do not have any GPU on travis so the CUDA drivers are not installed and `GEOSX` cannot be run:
```
[1002/1083] Generating XML schema to install
FAILED: CMakeFiles/geosx_generate_install_schema /tmp/build/CMakeFiles/geosx_generate_install_schema 
cd /tmp/build && /tmp/build/bin/geosx -s /tmp/build/schema.xsd >generate_schema.log 2>&1 || ( cat generate_schema.log && exit 1 )
GEOSX version: 0.2.0
  - c++ compiler: clang 10.0.0
  - cuda compiler version: 11.2
  - MPI version: Open MPI v4.0.3, package: Debian OpenMPI, ident: 4.0.3, repo rev: v4.0.3, Mar 03, 2020
  - HDF5 version: 1.12.1
...
CUDA ERROR (code = 35, CUDA driver version is insufficient for CUDA runtime version) at general.c:253
```
As options, we could
- Add a `cmake` option to control this `cmake` step. That implies to update `travis` as well, and I'm not sure this will be seamless to change the option in the `host-config`s only on `travis`.
- Ignore the error code (not my favorite option, obviously).
- Check for the `TRAVIS` environment variable to be present and set to `true` (see [here](https://docs.travis-ci.com/user/environment-variables/#default-environment-variables)), and deploy the schema accordingly.
- Define a `cmake` option to control this step, set to `true` by default for everyone, but overwritten to `false` if `TRAVIS` exists and is set to `true`.

If we chose the `cmake` option way, we'll surely have to update the doc as well.
(Not a big deal, but still needs to be done).

Reply:
This problem is CI related. We should not put a cmake ghost argument, CI specific, to resolve this CI issue IHMO.
What do you think of passing the option `--build-exe-only` to the `travis_build_and_test.sh` script in case of GPU related builds ? (as the green target `Ubuntu CUDA debug (20.04, clang 10.0.0 + gcc 9.4.0, open-mpi 4.0.3, cuda-11.2.152)`)

Reply:
It's obviously better to have the options properly defined in the build.
Here the option would be some `--deploy-scheme`. If you manage to make it happen, that's great.

I do not think we can rely on `--build-exe-only`: knowing that the tests can be built is important, even if we can't run them.

Reply:
Do you mean that the tests are build with option `--build-exe-only` ? If I understand the CMakeLists correctly they are build with the `all` target. 
What you want is to be able to (compile only the executable or executable and tests) and choose to install or not the schema as a separate step. Default behavior would install the schema. That's not so easy to implement only passing arguments to cmake to configure step.